lib/compat/wordpress-6.7/class-gutenberg-rest-templates-controller-6-7.php

```diff
@@ -0,0 +1,168 @@
+			list( , $slug ) = explode( '//', $request['id'] );
+			$template       = WP_Block_Templates_Registry::get_instance()->get_by_slug( $slug );
+			// @core-merge: End of changes to merge in core.
+		} else {
+			$template = get_block_template( $request['id'], $this->post_type );
+		}
+
+		if ( ! $template ) {
+			return new WP_Error( 'rest_template_not_found', __( 'No templates exist with that id.' ), array( 'status' => 404 ) );
+		}
+
+		return $this->prepare_item_for_response( $template, $request );
+	}
+
+	/**
+	 * Prepare a single template output for response
+	 *
+	 * @param WP_Block_Template $item    Template instance.
+	 * @param WP_REST_Request   $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$template = $item;
+
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( 'plugin' !== $item->origin ) {
+			return parent::prepare_item_for_response( $item, $request );
+		}
+		// @core-merge: Fix wrong author in plugin templates.
+		$cloned_item = clone $item;
+		// Set the origin as theme when calling the previous `prepare_item_for_response()` to prevent warnings when generating the author text.
+		$cloned_item->origin = 'theme';
+		$response            = parent::prepare_item_for_response( $cloned_item, $request );
+		$data                = $response->data;
+
+		if ( rest_is_field_included( 'origin', $fields ) ) {
+			$data['origin'] = 'plugin';
+		}
+
+		if ( rest_is_field_included( 'author_text', $fields ) ) {
+			$data['author_text'] = $this->get_wp_templates_author_text_field( $template );
+		}
+
+		if ( rest_is_field_included( 'original_source', $fields ) ) {
+			$data['original_source'] = $this->get_wp_templates_original_source_field( $template );
+		}
+
+		$response = rest_ensure_response( $data );
+
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$links = $this->prepare_links( $template->id );
+			$response->add_links( $links );
+			if ( ! empty( $links['self']['href'] ) ) {
+				$actions = $this->get_available_actions();
+				$self    = $links['self']['href'];
+				foreach ( $actions as $rel ) {
+					$response->add_link( $rel, $self );
+				}
+			}
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Returns the source from where the template originally comes from.
+	 *
+	 * @param WP_Block_Template $template_object Template instance.
+	 * @return string                            Original source of the template one of theme, plugin, site, or user.
+	 */
+	private static function get_wp_templates_original_source_field( $template_object ) {
+		if ( 'wp_template' === $template_object->type ) {
+			// Added by theme.
+			// Template originally provided by a theme, but customized by a user.
+			// Templates originally didn't have the 'origin' field so identify
+			// older customized templates by checking for no origin and a 'theme'
+			// or 'custom' source.
```

Comment:
Follow WPCS [formatting rules for multi-line comments](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-inline-comments). The same applies to both JS and PHP.

:1234: Applies throughout.

Reply:
Heads-up that this method (and several others) were copy-and-pasted from what's already in the WordPress codebase:

https://github.com/WordPress/wordpress-develop/blob/922949fb97334ec0bff551afb0bd8116ff8ade7c/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L807-L845

What's the correct way to proceed, here? Should I change the format of these comments here or directly in the WordPress codebase? Mentioning as this PR only includes this methods because they are `private`, but it isn't modifying anything.

Reply:
I think we could go either way, but since it's just comments, maybe update here and the backport the changes later.

It's quite easy to miss such subtle changes(comments) during backport and end up with the wrong format again 😄

Reply:
Makes sense, done in bbaab4dbf8365195daf7cd526058e443a06f85e2.